metashare/api/jobs.py

```diff
@@ -247,21 +184,27 @@ def get_unsaved_changes(scratch_org):
         if unsaved_changes:
             scratch_org.latest_revision_numbers = new_revision_numbers
             scratch_org.unsaved_changes = unsaved_changes
+    except Exception as e:
+        scratch_org.unsaved_changes = {}
+        async_to_sync(report_scratch_org_error)(
+            scratch_org, e, "SCRATCH_ORG_FETCH_CHANGES_FAILED"
+        )
+        tb = traceback.format_exc()
+        logger.error(tb)
+        raise
+    finally:
+        scratch_org.currently_refreshing_changes = False
```

Comment:
Would it be clearer to move this into `finalize_get_unsaved_changes`, so that both changes to `currently_refreshing_changes` live together in the model?